clinica/iotools/converters/ixi_to_bids/ixi_to_bids_utils.py

```diff
@@ -0,0 +1,209 @@
+import json
+import re
+import shutil
+from pathlib import Path
+from typing import List, Optional
+
+import nibabel as nib
+import pandas as pd
+from nilearn.image import concat_imgs
+
+from clinica.iotools.bids_utils import StudyName, bids_id_factory
+from clinica.utils.stream import cprint
+
+
+def get_subjects_list_from_data(data_directory: Path) -> List[str]:
+    return list(
+        dict.fromkeys(
+            re.match(r"IXI\d{3}", path.name).group(0)
+            for path in data_directory.rglob(pattern="IXI*.nii.gz")
+            if re.match(r"IXI\d{3}", path.name)
+        )
+    )
+
+
+def filter_subjects_list(
+    subjects_list: List[str], clinical_data: pd.DataFrame
+) -> List[str]:
+    return [
+        subject
+        for subject in subjects_list
+        if subject in clinical_data["ixi_id"].values
+    ]
+
+
+def define_participants(
+    data_directory: Path,
+    clinical_data: pd.DataFrame,
+    subjs_list_path: Optional[Path] = None,
+) -> List[str]:
+    """Define the actual list of participants based on the (provided) subjects list filtered
+    using clinical data"""
+    if subjs_list_path:
+        cprint("Loading a subjects list provided by the user...")
+        subjects_to_filter = subjs_list_path.read_text().splitlines()
+    else:
+        subjects_to_filter = get_subjects_list_from_data(data_directory)
+    return filter_subjects_list(subjects_to_filter, clinical_data)
+
+
+def read_ixi_clinical_data(clinical_data_path: Path) -> pd.DataFrame:
+    """Reads and formats IXI clinical data"""
+    # todo : what keys ?
+    clinical_data = pd.read_excel(clinical_data_path / "IXI.xls")
+    clinical_data.dropna(inplace=True)
+    clinical_data.drop("DATE_AVAILABLE", axis=1, inplace=True)
+    clinical_data.rename(lambda x: x.lower(), axis=1, inplace=True)
+    clinical_data["ixi_id"] = clinical_data.ixi_id.apply(
+        lambda x: "IXI" + "0" * (3 - len(str(x))) + str(x)
+    )
+    clinical_data["session_id"] = "ses-M000"
+    return clinical_data
+
+
+def rename_ixi_modalities(input_mod: str) -> str:
+    if input_mod == "T1":
+        return "T1w"
+    elif input_mod == "T2":
+        return "T2w"
+    elif input_mod == "MRA":
+        return "angio"
+    elif input_mod == "PD":
+        return "PDw"
+    elif input_mod == "DTI":
+        return "dti"
+    else:
+        raise ValueError(
+            f"The modality {input_mod} is not recognized in the IXI dataset."
+        )
+
+
+def define_magnetic_field(hospital: str) -> str:
+    if hospital == "Guys" or hospital == "IOP":
+        return "1.5"
+    if hospital == "HH":
+        return "3"
+
+
+def get_img_data_df(data_directory: Path) -> pd.DataFrame:
```

Comment:
You probably tired of me debating names 😅, but I think that we can afford the two missing letters to have a name easier to read : `get_image_data_df`. It doesn't sound like much but it contributes in reducing the cognitive load on the reader.